objects/Editors/contained/Reminder/contained/b2a49e/script.lua

```diff
@@ -0,0 +1,83 @@
+-- Gets the object (spirit or power card) beneath the reminder token
+function getObject()
+    local hits = Physics.cast({
+        origin = self.getBounds().center,
+        direction = Vector(0,-1,0),
+        max_distance = 2,
+    })
+    for _,v in pairs(hits) do
+        local obj = v.hit_object
+        if (obj.hasTag("Spirit") and obj.type == "Tile") or obj.type == "Card" then
+            return obj
+        end
+    end
+    return nil
+end
+
+-- Gets a set of parameters for image location suitable to be saved to script state
+-- These are agnostic of the actual image on the object, and of the mask size
+function getImageLocation(params)
+    local obj = params.obj
+    if obj == nil then
+        return nil
+    end
+
+    local location = {}
+
+    if obj.type == "Card" then
+        location.field = "FaceURL"
+    elseif obj.is_face_down then
+        location.field = "ImageSecondaryURL"
+    else
+        location.field = "ImageURL"
+    end
+
+    local selfPos = self.getPosition()
+    local objPos = obj.getPosition()
+    local selfBounds = self.getBounds()
+    local objBounds = obj.getBounds()
+    local selfSize = selfBounds.size.x -- We're not quite square, so only use our width
+
+    location.width = objBounds.size.x / selfSize
+    location.height = objBounds.size.z / selfSize
+
+    location.x = (objPos.x - selfPos.x) / objBounds.size.x * location.width
+    location.y = (objPos.z - selfPos.z) / objBounds.size.z * location.height
+
+    return location
+end
+
+-- Sets this object's position and size above a given object such that getImageLocation() would return a given location.
+-- We can't do this with takeObject(), as that can't set size.
+-- So we might as well do this here, so that the code sits next to getImageLocation().
+function setToLocation(params)
+    local obj = params.obj
+    local location = params.location
+    if obj == nil or location == nil then
+        return
+    end
+
+    local objPos = obj.getPosition()
+    local selfBounds = self.getBounds()
+    local objBounds = obj.getBounds()
+    local selfSize = selfBounds.size.x
+
+    local desiredX = objPos.x - location.x * objBounds.size.x / location.width
+    local desiredZ = objPos.z - location.y * objBounds.size.z / location.height
+    local desiredY = objPos.y + 1
+    self.setPosition(Vector(desiredX, desiredY, desiredZ))
+
+    local desiredSize = objBounds.size.x / location.width
+    self.setScale(self.getScale():scale(desiredSize / selfSize))
```

Comment:
not sure if this matters, but aren't you also scaling the Y here as well?

Reply:
Yes, I'm intentionally scaling the X and Y both according to a scaling factor calculated from the X alone. I previously tried scaling the X based on the X and the Y based on the Y, but `getBounds().size` returns slightly different X and Y sizes (due to the wobbly nature of the edges of the reminders), so that made things non-square.

I actually put a comment to this effect in `getImageLocation()`, and `setToLocation()` uses the inverse of the maths in `getImageLocation()`, but I should probably put the same comment in `setToLocation()` just to make sure.

Reply:
the Y is the height of the token, but didn't realize that when you hit +/- it also scaled the Y for the tokens, guess it's just Cards that are stuck at always 1.0

Reply:
Oh, right, Y, not Z. Yeah, not necessary, but as you say, more in line with what the player will be doing.